### DISCUSSION TITLE
How to add own workspace

### DISCUSSION THREAD

**Discussion by Malher120** [hajkmap/Hajk#1466] [Q&A]
I've tried to make my own workspace from my own local geoserver however when I try to get the workspaces I get an error message and dont know what Im doing wrong. Attaching images of my Hajk admin and my geoserver workspace
![HajkAsk](https://github.com/hajkmap/Hajk/assets/158027688/f17e2f63-02af-49cc-823e-72acea456bd4)
![HajkAsk2](https://github.com/hajkmap/Hajk/assets/158027688/23ed5a89-2254-4bb3-9647-59553c9d12d8)
![Untitled](https://github.com/hajkmap/Hajk/assets/158027688/6468f659-c3ac-462d-b720-2fa8686b559e)

**Comment by Hallbergs** [comment]
Could you take a look at the network tab in the browser when clicking the "Get workspace" button? My guess is that your Geoserver rest-api isn't enabled.

**Reply by Malher120** [reply]
I see that the request url is wrong and changed from what I put as an input. 
![Untitled](https://github.com/hajkmap/Hajk/assets/158027688/0e7bcb23-704a-4903-8bf2-e4e50d08416e)
![Untitled1](https://github.com/hajkmap/Hajk/assets/158027688/5735ac89-ed7c-4b8d-8f31-f148c39c02f5)

**Comment by Hallbergs** [comment]
The code in the admin-application is a bit of a mess, but if i read it correctly i think you will get the correct url if you add `/` at the end of your url!

**Comment by Malher120** [comment]
I get this then, still seems very weird. Thing is, I have a local geoserver up and running but when I put that into the url I dont get anything either, just the wrong utl

![image](https://github.com/hajkmap/Hajk/assets/158027688/88dfc283-0a9c-4edf-84d3-c8d96dd63a92)

**Comment by Hallbergs** [comment]
Ah, my bad. I thought the "naturalearthdata"-url was pointing to your geoserver. You should use your geoserver url in the admin-application. If you're running it locally it is probably something like http://localhost:8080/geoserver/. How does the request/response look if you use the url to your local geoserver?

**Reply by Malher120** [reply]
I get a 401 then. How do I authorize Hajk to access my geoserver?

![image](https://github.com/hajkmap/Hajk/assets/158027688/3a86e30c-9873-4dce-8a2e-1bded92ab9dc)

**Reply by Hallbergs** [reply]
If you're running it locally you can just set the access rules in geoserver to allow "everything". Otherwise you have to run the requests trough a proxy which can add authentication information to your request. There is a geoserver-proxy under /proxy in the repo, i have never used it so i'm afraid i wont be of much help regarding the proxy.

**Reply by Malher120** [reply]
I tried setting every data access setting to allow everything but still get the same error

![image](https://github.com/hajkmap/Hajk/assets/158027688/8a77f7cd-820a-4288-8d3c-84ff0ebf82df)
![image](https://github.com/hajkmap/Hajk/assets/158027688/90b21d89-55fe-4e6d-8600-2413523a33c5)
![image](https://github.com/hajkmap/Hajk/assets/158027688/27a23fb7-b0f7-4db3-9acb-303504b610b5)
